### ISSUE TITLE
Array_Info and Telescope_Info are empty in some h5 files

### ISSUE THREAD

**Issue by mikael10j** [cta-observatory/dl1-data-handler#24]
For some h5 files on the grid, Array_Info and Telescope_Info tabs are empty.
I've also noticed that some file are really small (sevel MB). Is there any reason for that ?
Thanks

**Comment by nietootein** [comment]
Hi Mikael! Thanks for your feedback. Would you mind telling us if those files belong to the DL1_ML1 or the ProtoML datasets? Posting the filename of any of the affected files may help us figure out if there could have been issues while reducing those data.

**Comment by mikael10j** [comment]
Hi Daniel,

**Comment by bryankim96** [comment]
@nietootein Do we still have any of the command line output logs from when these runs were processed? Maybe it will give some clue as to what is going on here. I think I also saw a couple of these unusual "empty" files. I initially thought it might have something to do with weird cut behavior, but now that I think about it again I don't believe we applied cuts for these datasets.

**Comment by nietootein** [comment]
Hi @bryankim96,

We continued this discussion on [Redmine](https://forge.in2p3.fr/issues/24661) with the people in charge of the data reduction on the grid. In summary, it seems that the most likely cause for the corrupted files that @mikael10j  noticed is an I/O error during processing (we infer this from the logs). Thus, it does not seem that the problem comes from image-extractor. Anyway, let's keep an eye on this should it happen again in the future. 

@mikael10j, could you throw us an approximate percentage of affected files over the entire dataset you are using?

D.

**Comment by bryankim96** [comment]
I'm not sure if this is strictly an improvement, but I made a change in 1a0e0908c070e27dca2eb07d88a500f70bf354a3 so that if there is an error/exception raised during processing of a data file and it stops before completion, the incomplete file will be removed. This should at least remove the issue of incomplete/corrupted files.

**Comment by mikael10j** [comment]
Hi there, 

I checked and I only found this file. But there were also several (~10) small files (<10 MB) not containing LSTs that I don't remember the name (I've already deleted them on my side). 

Cheers, 

Mikaël 


De: "Bryan Kim" <notifications@github.com> 
À: "cta-observatory/image-extractor" <image-extractor@noreply.github.com> 
Cc: "mikael10j" <mikael.jacquemont@lapp.in2p3.fr>, "Mention" <mention@noreply.github.com> 
Envoyé: Vendredi 28 Septembre 2018 08:19:43 
Objet: Re: [cta-observatory/image-extractor] Array_Info and Telescope_Info are empty in some h5 files (#24) 



I'm not sure if this is strictly an improvement, but I made a change in [ https://github.com/cta-observatory/image-extractor/commit/1a0e0908c070e27dca2eb07d88a500f70bf354a3 | 1a0e090 ] so that if there is an error/exception raised during processing of a data file and it stops before completion, the incomplete file will be removed. This should at least remove the issue of incomplete/corrupted files.
